hazelcast/src/main/java/com/hazelcast/jet/impl/operation/UploadJobMultiPartOperation.java

```diff
@@ -44,19 +49,34 @@ public UploadJobMultiPartOperation(JetUploadJobMultipartCodec.RequestParameters
         jobMultiPartParameterObject.setSha256Hex(parameters.sha256Hex);
     }
 
+    @Override
+    public Object getResponse() {
+        return response;
+    }
 
     @Override
-    public CompletableFuture<Boolean> doRun() {
-        return CompletableFuture.supplyAsync(() -> {
```

Comment:
Won't it be sufficient to just use second parameter here with other ExecutionService, e.g. CALLER_RUNS?

Reply:
Actually in the beginning @frant-hartm had asked me to only inherit from Operation class. I did it for UploadJobMetaDataOperation but somehow I could not do it for UploadJobMultiPartOperation. So UploadJobMultiPartOperation was still inheriting from AsyncJobOperation. 
Then I realized that I could inherit from Operation class. So it is consistent with his comment now

Reply:
Yes, I prefer it this way actually.

But I would like to know there was the exception previously.